### PR TITLE
Don’t remove whitespace inside the string

### LINKED ISSUE #1511: Version bounds with a space not a comma
I made the mistake of writing this
```
julia = "0.7 1.0"
```
over [here](https://github.com/JuliaMath/VML.jl/pull/24/commits/1e2c6200b405d44ee879c034b1d7daa05cd37a2f), which leads to warnings when loading the package on Julia 1.3.

@fredrikekre points out that this is parsed as follows:
```
julia> Pkg.Types.semver_spec("0.7 1.0")
VersionSpec("0.71”)
```
Surely I’m not the only guy who’ll ever make this mistake. Perhaps it should either be an error, or be accepted?

### PR DESCRIPTION
**Pull request by DilumAluthge**
Fixes #1511

### CODE REVIEW

File: src/versions.jl [hunk 1]
```diff
@@ -223,7 +223,7 @@ function Base.intersect(A::VersionSpec, B::VersionSpec)
         i += 1
     end
     VersionSpec(ranges)
-end
+endi
```
Comment (fredrikekre) [inline src/versions.jl]: 🤔

Comment (DilumAluthge) [reply src/versions.jl]: GitHub web editor + iPad on-screen keyboard.

Comment (DilumAluthge) [reply src/versions.jl]: Actually it’s my new syntax I’m proposing for Julia 2.0. Deprecate `end` in favor of `endi`.

**Comment by DilumAluthge** [comment]
Alright, this is ready to go. @fredrikekre @KristofferC 

As you can see from the new test, "0.7 1.0" et cetera will now throw an error.

**Comment by DilumAluthge** [comment]
Bump. Any opposition to merging this?

**Comment by KristofferC** [comment]
Thanks!

**Pull request merged**